### PR TITLE
Kernel cache drop

### PR DESCRIPTION
**Pull request by bengland2**
This cherrypy-based web service allows ripsaw benchmarks to request a kernel cache drop from a privileged pod without themselves being privileged.     The DaemonSet yaml will be part of ripsaw so that any storage benchmark can start and invoke this service.   Tested with minikube v1.11.0 = k8s v1.18.

### CODE REVIEW

File: kernel_cache_dropper/kernel-cache-drop-websvc.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            logger.info('completed cache drop')
+        except FileNotFoundError:
+                logger.error('/proc_sys_vm/drop_caches not found')
+        return 'SUCCESS'
+
+if __name__ == '__main__':
+    config = { 
+        'global': {
+            'server.socket_host': '0.0.0.0' ,
+            'server.socket_port': 9435,
```
Comment (jtaleric) [inline kernel_cache_dropper/kernel-cache-drop-websvc.py]: Want to take this in as a env var that way the user can overwrite it? Default to 9435, but allow user to provide a means to overwrite?

**Review by jtaleric — APPROVED** [review]
LGTM, just one suggestion, not worth blocking.

File: kernel_cache_dropper/kernel-cache-drop-websvc.py (hunk 1 above)
Comment (bengland2) [reply kernel_cache_dropper/kernel-cache-drop-websvc.py]: I agree, had started that but haven't finished it yet.

File: kernel_cache_dropper/kernel-cache-drop-websvc.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    def DropKernelCache(self):
+        logger.info('asked for cache drop')
+        try:
+            os.sync()
+            logger.info('completed sync call')
+            with open('/proc_sys_vm/drop_caches','a') as dcf:
+                dcf.write('3\n')
+            logger.info('completed cache drop')
+        except FileNotFoundError:
+                logger.error('/proc_sys_vm/drop_caches not found')
```
Comment (rsevilla87) [inline kernel_cache_dropper/kernel-cache-drop-websvc.py]: This one is returning 200 OK even if /proc_sys_vm/drop_caches doesn't exist.

**Review by rsevilla87 — CHANGES_REQUESTED** [review]
I think we should verify that the script is running as root, or handle that exception.

File: kernel_cache_dropper/kernel-cache-drop-websvc.py (hunk 2 above)
Comment (bengland2) [reply kernel_cache_dropper/kernel-cache-drop-websvc.py]: @rsevilla87  wow, it never occurred to me that it wouldn't exist.   I was only concerned that the pod wouldn't have access to it.   But if the file doesn't exist, then the exception won't be caught and the program will exit.    Are you sure that you passed the pod IP to it?   request_cache_drop.py does absolutely nothing unless it has pod IPs passed to it.

Comment (rsevilla87) [reply kernel_cache_dropper/kernel-cache-drop-websvc.py]: It always "return 'SUCCESS'" regardless a exception is caught or not :)

Comment (jtaleric) [reply kernel_cache_dropper/kernel-cache-drop-websvc.py]: I think the suggestion is just to pass a 404 if we hit the `FileNotFoundError`

**Review by rsevilla87 — APPROVED** [review]

**Comment by rsevilla87** [comment]
LGTM!

**Pull request merged**